velox/dwio/common/SelectiveColumnReader.cpp

```diff
@@ -220,6 +220,54 @@ void SelectiveColumnReader::getIntValues(
   }
 }
 
+void SelectiveColumnReader::getUnsignedIntValues(
+    RowSet rows,
+    const TypePtr& requestedType,
+    VectorPtr* result) {
+  switch (requestedType->kind()) {
+    case TypeKind::SMALLINT:
+      switch (valueSize_) {
+        case 4:
+          getFlatValues<int32_t, int16_t>(rows, result, requestedType);
+          break;
+        default:
+          VELOX_FAIL("Unsupported value size: {}", valueSize_);
+      }
+      break;
+    case TypeKind::INTEGER:
+      switch (valueSize_) {
+        case 4:
+          getFlatValues<int32_t, int32_t>(rows, result, requestedType);
+          break;
+        default:
+          VELOX_FAIL("Unsupported value size: {}", valueSize_);
+      }
+      break;
+    case TypeKind::BIGINT:
+      switch (valueSize_) {
+        case 4:
+          getFlatValues<uint32_t, int64_t>(rows, result, requestedType);
+          break;
+        default:
+          VELOX_FAIL("Unsupported value size: {}", valueSize_);
+      }
+      break;
+    case TypeKind::HUGEINT:
+      switch (valueSize_) {
+        case 8:
```

Comment:
is valueSize_ = 16 a valid case?

Reply:
@yingsu00  thank you, the PR mainly focus PR unsigned int support, and for Parquet unsigned int, the parquet store 8/16/32 with bitswidth 32, for uint64 it store with bitswidth 64. So only 4 and 8 are valid case here.

Reply:
@gaoyangxiaozhu I think it's a valid case. Can you please add case 16 here?

Reply:
sure @yingsu00